presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -65,7 +65,7 @@ public TypedSet(Type elementType, int expectedSize)
 
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + elementBlock.getRetainedSizeInBytes() + blockPositionByHash.sizeOf();
+        return INSTANCE_SIZE + elementBlock.getRetainedSizeInBytes() + blockPositionByHash.size() * Integer.BYTES;
```

Comment:
Get the INSTANCE_SIZE of IntArrayList using JOL. You can find an example at the beginning at this class. Add that into the retained size.